DigitalLearningSolutions.Web/Controllers/LoginController.cs

```diff
@@ -69,50 +71,85 @@ public IActionResult Index(LoginViewModel model)
-            return View("ChooseACentre", chooseACentreViewModel);
+            SetTempDataForChooseACentre
+            (
+                model.RememberMe,
+                adminClaimDetails,
+                delegateClaimDetails,
+                chooseACentreViewModel
+            );
+
+            return RedirectToAction("ChooseACentre", "Login");
         }
 
+        [ServiceFilter(typeof(RedirectEmptySessionData<ChooseACentreViewModel>))]
         [HttpGet]
-        public IActionResult ChooseACentre(int centreId)
+        [ResponseCache(Location = ResponseCacheLocation.None, NoStore = true)]
+        public IActionResult ChooseACentre()
+        {
+            if (User.Identity.IsAuthenticated)
+            {
+                return RedirectToAction("Index", "Home");
+            }
+
+            var model = TempData.Peek<ChooseACentreViewModel>();
+
+            return View("ChooseACentre", model);
+        }
+
+        [ServiceFilter(typeof(RedirectEmptySessionData<List<DelegateUser>>))]
+        [HttpGet]
+        public IActionResult ChooseCentre(int centreId)
         {
             var rememberMe = (bool)TempData["RememberMe"];
-            var adminAccount = TempData.Peek<AdminUser>();
-            var approvedDelegateAccounts = TempData.Peek<List<DelegateUser>>();
+            var adminClaimDetails = TempData.Peek<AdminClaimDetails>();
+            var delegateClaimDetails = TempData.Peek<List<DelegateClaimDetails>>();
             TempData.Clear();
 
-            var adminAccountForChosenCentre = adminAccount?.CentreId == centreId ? adminAccount : null;
+            var adminAccountForChosenCentre = adminClaimDetails?.CentreId == centreId ? adminClaimDetails : null;
             var delegateAccountForChosenCentre =
-                approvedDelegateAccounts?.FirstOrDefault(du => du.CentreId == centreId);
+                delegateClaimDetails?.FirstOrDefault(du => du.CentreId == centreId);
             if (adminAccountForChosenCentre != null)
             {
                 sessionService.StartAdminSession(adminAccountForChosenCentre.Id);
             }
+
             return LogIn(adminAccountForChosenCentre, delegateAccountForChosenCentre, rememberMe);
         }
 
-        private void SetTempDataForChooseACentre(bool rememberMe, AdminUser? adminUser,
-            List<DelegateUser> approvedDelegateUsers)
+        private (AdminClaimDetails?, List<DelegateClaimDetails>) GetClaimsDetails(AdminUser? adminUser, List<DelegateUser> delegateUsers)
+        {
+            var adminClaimDetails = adminUser != null ? new AdminClaimDetails(adminUser) : null;
+            var delegateClaimDetails = delegateUsers.Select(du => new DelegateClaimDetails(du)).ToList();
+            return (adminClaimDetails, delegateClaimDetails);
+        }
+
+        private void SetTempDataForChooseACentre(bool rememberMe, AdminClaimDetails? adminClaimDetails,
```

Comment:
If we need to split the params on new lines, could we add each of them on a new line for better readability? 🙏  (I don't think the formatter does it by default)